### ISSUE TITLE
"Actually" random random play

### ISSUE THREAD

**Issue by dvx** [dvx/lofi#37]
This has been asked for a while now, but this would be a good rundown of a "random play" feature:

(from https://news.ycombinator.com/item?id=20346406, by @patmcc)

1. I want it to play in a truly random fashion, with replacement (same song twice in a row is possible).

2. I want it to play in a truly random fashion, without replacement (each song is removed from the list once played).

3. I want it to "seem" random (don't play too many songs by the same artist in a row).

4. I want it to be random, but predictable (same "random" ordering if I start the playlist again).

5. I want it to play my favourite songs more often than my lower-rated songs.

6. I want it to play my less-commonly played songs more often than the stuff I listen to all the time.

**Comment by tonyreta03** [comment]
I just found the app and I love the concept of the app, I do wish that I could resize it so I will start working on that along with the resizable concept.

**Comment by dvx** [comment]
Hey @tonyreta03 -- do you mind working off of my `random-shuffle` branch? https://github.com/dvx/lofi/tree/random-shuffle

I've done most of the UI work but I'm still thinking of how exactly shuffling should be done. More than happy to discuss an ideal solution.

**Comment by dvx** [comment]
I was thinking the best way to handle it would be to create a "shuffled" playlist. This make sure that the "shuffle" persists through devices. But unfortunately, this won't work with just playing a radio playlist.

Anyway, just a thought.

**Comment by tonyreta03** [comment]
@dvx yeah I'd be more than happy to work on that branch, i am currently looking at college stuff right this minute but I will continue working on advancements as soon as I finish.

**Comment by dvx** [comment]
@tonyreta03 -- No rush, thanks for your interested in the project! :D
